src/app/common/ResourceCodeBlock.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <StatusInfo
+    :is-loading="isLoading"
+    :error="error"
+  >
+    <CodeBlock
+      :id="id"
+      language="yaml"
+      :code="yamlUniversal"
+      :is-searchable="props.isSearchable"
+      :query-key="props.id"
+      :code-max-height="props.codeMaxHeight"
+    >
+      <template #secondary-actions>
+        <KButton
+          appearance="outline"
+          size="small"
+          :is-rounded="false"
+          :title="i18n.t('common.copyKubernetesText')"
+          @click="copyAsKubernetes"
+        >
+          <KIcon
+            color="currentColor"
+            icon="copy"
+            size="18"
+            :title="i18n.t('common.copyKubernetesText')"
+          />
+
+          {{ i18n.t('common.copyKubernetesShortText') }}
+        </KButton>
```

Comment:
We discussed in GUI sync whether this should re-use the reusable CopyButton component from https://github.com/kumahq/kuma-gui/pull/897. Personally I think it should whether here or in a quick follow up.

Reply:
I’ve looked into this and it’s not quite as straightforward.

The "Copy as Kubernetes" button’s action fetches the Kubernetes format of the resource on demand. However, in order to use said CopyButton component, we need to know the text that’s to be copied up-front. We could fetch the Kubernetes version up front, but that’s a waste of network resources. Alternatively, CopyButton could have a prop `getText` which returns the text to be copied.